jib-cli/src/main/java/com/google/cloud/tools/jib/cli/jar/JarModeProcessor.java

```diff
@@ -123,7 +127,10 @@ static JarType determineJarType(Path jarPath) throws IOException {
         snapshotDependencies.forEach(
             path ->
                 snapshotDependenciesLayerBuilder.addEntry(
-                    path, APP_ROOT.resolve(RelativeUnixPath.get("dependencies")).resolve(path)));
+                    jarParent.resolve(path),
+                    APP_ROOT
+                        .resolve(RelativeUnixPath.get("dependencies"))
```

Comment:
```suggestion
                        .resolve("dependencies")
```
and the one above tool.